cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/templates/slurm_parallelcluster_queue_partition.conf

```diff
@@ -1,38 +1,89 @@
+{%- import 'common/slurm_utils.conf' as slurm_utils -%}
+
 # This file is automatically generated by pcluster
-{% set ns = namespace(has_node=false) %}
+{% set ns = namespace(has_node=false) +%}
 
-{% for compute_resource in queue_config.ComputeResources %}
-    {% set static_size = compute_resource.MinCount %}
-    {% set dynamic_size = compute_resource.MaxCount - compute_resource.MinCount %}
-    {% set instance_type = compute_resource.InstanceType%}
-    {% if static_size > 0 %}
-NodeName={{ queue_name }}-st-{{ compute_resource.Name }}-[1-{{ static_size }}] CPUs={{ compute_resource | vcpus }} RealMemory={{ compute_resource | realmemory }} State=CLOUD Feature=static,{{ instance_type }},{{ compute_resource.Name }}{% if compute_resource.Efa.Enabled %},efa{% endif %}{% if instance_type | gpus > 0 %},gpu{% if not no_gpu %} Gres=gpu:{{ instance_type | gpu_type }}:{{ instance_type | gpus }}{% endif %}{% endif %}
+{#- Macros: #}
+{%- macro cpus(compute_resource) -%}
```

Comment:
why not move this into the utils.conf file?

Reply:
I moved only macros that are common between the queue/partition & queue/gres configuration files.

I'll go ahead and move these macros as well to the utils file to improve the readability of the actual configuration files.